Source/DataModels/Resources/ResourceNavMesh.h

```diff
@@ -0,0 +1,322 @@
+#pragma once
+#include "Resource.h"
+//#include "Recast/SampleInterfaces.h"
+//#include "Utils/Buffer.h"
+#include <vector>
+
+
+class ResourceNavMesh : virtual public Resource
+{
+public:
+	ResourceNavMesh();
+	virtual ~ResourceNavMesh() override;
+
+	//bool Build(Scene* scene);
+	//void DrawGizmos(Scene* scene);	
+	//void Load(Buffer<char>& buffer);
+	//void CleanUp();
+	//Buffer<char> Save();
+
+	//bool IsGenerated() const;	
+	//dtCrowd* GetCrowd() const;
+	//dtNavMeshQuery* GetNavMeshQuery() const;
+	//dtNavMesh* GetNavMesh() const;
+	//dtTileCache* GetTileCache() const;
+
+	float GetAgentHeight() const;
+	float GetAgentRadius() const;
+	float GetAgentMaxClimb() const;
+	float GetAgentMaxSlope() const;
+	float GetCellSize() const;
+	float GetCellHeight() const;
+	float GetEdgeMaxError() const;
+
+	void SetAgentHeight(float agentHeight);
+	void SetAgentRadius(float agentRadius);
+	void SetAgentMaxClimb(float agentMaxClimb);
+	void SetAgentMaxSlope(float agentMaxSlope);
+	void SetCellSize(float cellSize);
+	void SetCellHeight(float cellHeight);
+	void SetEdgeMaxError(float edgeMaxError);
+
+	int GetRegionMinSize() const;
+	int GetRegionMergeSize() const;
+	int GetPartitionType() const;
+	int GetEdgeMaxLen() const;
+	int GetVertsPerPoly() const;
+	int GetDetailSampleDist() const;
+	int GetDetailSampleMaxError() const;
+	int GetTileSize() const;
+	int GetMaxTiles() const;
+	int GetMaxPolysPerTile() const;
+
+	void SetRegionMinSize(int regionMinSize);
+	void SetRegionMergeSize(int regionMergeSize);
+	void SetPartitionType(int partitionType);
+	void SetEdgeMaxLen(int edgeMaxLen);
+	void SetVertsPerPoly(int vertsPerPoly);
+	void SetDetailSampleDist(int detailSampleDist);
+	void SetDetailSampleMaxError(int detailSampleMaxError);
+	void SetTileSize(int tileSize);
+	void SetMaxTiles(int maxTiles);
+	void SetMaxPolysPerTile(int maxPolysPerTile);
+
+	bool GetKeepInterResults() const;
+
+	void SetKeepInterResults(bool keepInterResults);
+
+	unsigned char GetNavMeshDrawFlags() const;
+	
+	void SetNavMeshDrawFlags(unsigned char navMeshDrawFlags);
+
+public:
+	enum class DrawMode
+	{
+		DRAWMODE_NAVMESH,
+		DRAWMODE_NAVMESH_TRANS,
+		DRAWMODE_NAVMESH_BVTREE,
+		DRAWMODE_NAVMESH_NODES,
+		DRAWMODE_NAVMESH_PORTALS,
+		DRAWMODE_NAVMESH_INVIS,
+		DRAWMODE_MESH,
+		DRAWMODE_CACHE_BOUNDS,
+		MAX_DRAWMODE
+	};
+
+//private:
+	//void InitCrowd();
+
+private:
+
+	float agentHeight = 2.0f;
+	float agentRadius = 0.5f;
+	float agentMaxClimb = 0.9f;
+	float agentMaxSlope = 45.0f;
+
+	float cellSize = 0.30f;
+	float cellHeight = 0.20f;
+
+	int regionMinSize = 8;
+	int regionMergeSize = 20;
+
+	int partitionType = 0;
+
+	int edgeMaxLen = 12;
+	float edgeMaxError = 1.3f;
+	int vertsPerPoly = 6;
+
+	int detailSampleDist = 6;
+	int detailSampleMaxError = 1;
+
+	bool keepInterResults = false;
+
+	int tileSize = 48;
+	int maxTiles = 0;
+	int maxPolysPerTile = 0;
+
+	DrawMode drawMode = DRAWMODE_NAVMESH;
```

Comment:
Looks like the pipeline is failing, try scoping the enum value, that should fix it.
```suggestion
	DrawMode drawMode = DrawMode::DRAWMODE_NAVMESH;
```